### PR TITLE
[android] Fix broken PICK_POINT API

### LINKED ISSUE #4780: Broken "Area selection" functionality
1. Create Intent for "pick point" feature:

```kotlin
    private val request = Intent(Intent.ACTION_VIEW, Uri.parse("om://crosshair?appname=DEMO TITLE"))
        .apply {
            putExtra("app.organicmaps.api.extra.PICK_POINT", true)
        }
```

2. Use `startActivityForResult()`
<details>
<summary>The app shows correct screen</summary>

image

</details>

3. Select an aria ant tap to accept button

Expected result: the caller received the coordinates and zoom
Actual result: `ActivityResult{resultCode=RESULT_CANCELED, data=null}`

### PR DESCRIPTION
**Pull request by rtsisyk**
Fixes #4780

### CODE REVIEW

File: android/src/app/organicmaps/DownloadResourcesLegacyActivity.java [hunk 1]
```diff
@@ -347,7 +356,7 @@ public void showMap()
     final Intent intent = new Intent(this, MwmActivity.class);
 
     // Disable animation because MwmActivity should appear exactly over this one
-    intent.addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION | Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_NEW_TASK);
```
Comment (biodranik) [inline android/src/app/organicmaps/DownloadResourcesLegacyActivity.java:350]: This reverts the fix for the crash https://github.com/organicmaps/organicmaps/issues/4240

Should it be really reverted? Why? A detailed comment in the code here would be helpful for anyone who tries to fix crashes again using the same approach.
And if it introduces crashes again, then the crash issue should be reopened. And it also has high priority, like a release blocker.

Comment (rtsisyk) [reply android/src/app/organicmaps/DownloadResourcesLegacyActivity.java:350]: > 

`FLAG_ACTIVITY_NEW_TASK` creates a new task and this breaks propagation of the activity result from MwmActivity to DownloadLegacyActivity -> SplashActivity.

Comment (biodranik) [reply android/src/app/organicmaps/DownloadResourcesLegacyActivity.java:350]: Did you check if #4240 reproduces again on this branch?

Comment (rtsisyk) [reply android/src/app/organicmaps/DownloadResourcesLegacyActivity.java:350]: No, #4240 was fixed by #4656.

**Review by biodranik — APPROVED** [review]
It would be great to have some tests that will check if API works.

**Pull request merged**